sources/ClangSharp/Interop/clangsharp.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft and Contributors. All rights reserved. Licensed under the University of Illinois/NCSA Open Source License. See LICENSE.txt in the project root for license information.
+
+// Ported from https://github.com/llvm/llvm-project/tree/llvmorg-9.0.0/clang/include/clang-c
+// Original source is Copyright (c) the LLVM Project and Contributors. Licensed under the Apache License v2.0 with LLVM Exceptions. See NOTICE.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace ClangSharp.Interop
+{
+    public static unsafe partial class clangsharp
+    {
+        private const string libraryPath = "libClangSharp";
+
+        [DllImport(libraryPath, EntryPoint = "clangsharp_Cursor_getBinaryOpcode", CallingConvention = CallingConvention.Cdecl)]
+        public static extern CX_BinaryOperatorKind Cursor_getBinaryOpcode(CXCursor C);
+
+        [DllImport(libraryPath, EntryPoint = "clangsharp_Cursor_getBinaryOpcodeStr", CallingConvention = CallingConvention.Cdecl)]
+        public static extern CXString Cursor_getBinaryOpcodeStr(CX_BinaryOperatorKind Op);
+
+        [DllImport(libraryPath, EntryPoint = "clangsharp_Cursor_getUnaryOpcode", CallingConvention = CallingConvention.Cdecl)]
+        public static extern CX_UnaryOperatorKind Cursor_getUnaryOpcode(CXCursor C);
+
+        [DllImport(libraryPath, EntryPoint = "clangsharp_Cursor_getUnaryOpcodeStr", CallingConvention = CallingConvention.Cdecl)]
+        public static extern CXString Cursor_getUnaryOpcodeStr(CX_UnaryOperatorKind Op);
+
+        [DllImport(libraryPath, EntryPoint = "clangsharp_getCursorExtent", CallingConvention = CallingConvention.Cdecl)]
+        public static extern CXSourceRange getCursorExtent(CXCursor param0);
```

Comment:
The `clang_getCursorExtent` API keeps the `start` location "as-is". However, it transforms the `end` location to its expansion location. This means that if the actual extent is in a macro, you get `macro start location, expansion end location`.

That transformation is what was blocking the token processing logic from working in some cases (namely when macros were involved).

This API just gives you the raw extent (`macro start location, macro end location`) and the consumer, if desired, can manually transform each location to the expansion location or any other location as needed. This allows token processing to work as needed and allows the consumer to customize as necessary.